packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,239 @@
 import type { CSSObject } from 'styled-components';
-import type { BaseBoxProps } from './types';
-import { getIn, makeBorderSize, makeSize, makeSpace } from '~utils';
+import type {
+  BaseBoxProps,
+  MakeValueResponsive,
+  SpacingValueType,
+  ArrayOfMaxLength4,
+} from './types';
+import type { Breakpoints } from '~tokens/global/breakpoints';
+import breakpoints from '~tokens/global/breakpoints';
+import { getMediaQuery } from '~src/utils/getMediaQuery';
+import { isReactNative, isEmpty, getIn, makeSpace, makeSize } from '~utils';
 import type { Theme } from '~components/BladeProvider';
 
-// allows 'auto' as spacing value
-const getSpacingValue = <SpacingValue extends string | number | undefined>({
-  value,
-  theme,
-}: {
-  value: SpacingValue;
-  theme: Theme;
-}): string | undefined => {
-  if (value === 'auto') return 'auto';
-  /**
-   * NOTE: don't allow numbers once we have the scale for sizing
-   * DO NOT PUBLISH THIS COMPONENT PUBLICLY WITH THIS PIECE OF CODE
-   */
-  if (typeof value === 'number') {
-    return makeSpace(value);
+const getResponsiveValue = <T extends string | number | string[]>(
+  value: MakeValueResponsive<T> | undefined,
```

Comment:
These are utility functions so the undefined or falsy values etc are handled inside this function and not before this function

Reply:
Making it optional is not what is expected here. 

We don't want to support this - `getResponsiveValue()` 
We just want to support - `getResponsiveValue(undefined)`

Although it will work for second case as well but don't see any reason to make prop itself optional